Lib/fontbakery/specifications/googlefonts.py

```diff
@@ -4799,5 +4799,34 @@ def com_google_fonts_test_165(ttFont, familyname):
                   "Please report this issue at:\n{}").format(url,
                                                              FB_ISSUE_TRACKER)
 
+
+@register_test
+@test(
+    id='com.google.fonts/test/166'
+  , rationale = """
+      The git sha1 tagging and dev/release features of Source Foundry font-v
+       tool are awesome and we would love to consider upstreaming the approach
+       into fontmake someday. For now we only emit a WARN if a given font does
+       not yet follow the experimental versionin style, but at some point we
+       may start enforcing it.
+    """
+  , request = 'https://github.com/googlefonts/fontbakery/issues/1563'
+)
+def com_google_fonts_test_166(ttFont):
+  """ Check for font-v versioning """
+  from fontv.libfv import FontVersion
+
+  fv = FontVersion(ttFont.reader.file.name)
+  if fv.version and (fv.is_development or fv.is_release):
+    yield PASS, "Font version string looks GREAT!"
+  else:
+    yield INFO, ("Version string is: \"{}\"\n"
+                 "The version string must ideally include a git commit hash"
+                 " and eigther a 'dev' or a 'release' suffix such as in the"
+                 " example below:\n"
+                 "\"Version 1.3; git-0d08353-release\"").format(fv.version)
```

Comment:
`fv.version` (in string formatting line) will currently give you **only** the version number substring (i.e. `Version X.XXX`).  Recommend changing the `fv.version` here to `fv.get_version_string()`.  The method returns full semicolon delimited version string with all included metadata.

Reply:
Pseudo code for this block is the following:

if there is a version number substring AND there is a substring at position 2 (after first semicolon) that has one of the following formats:

- `DEV`
- `RELEASE`
- `[sha1]-dev`
- `[sha1]-release`

THEN pass test

Reply:
your pseudo-code is merely explaining what I did? Or do you think I did it wrong and this is a suggestion for a rewrite?

Reply:
No, was trying to interpret the object attribute definitions in pseudo code to confirm that it is clear what you are testing here.  `version` will be a version number substring as `Version X.XXX` and `is_development` and `is_release` will take the above definition.  The logic looks good to me but I wanted to clarify that you understood the test that you were performing because you included `fv.version` for string formatting and that will not print the full version string, only the version number substring.  Assumed that you believed you were getting the entire version string with that attribute.